core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2541,6 +2547,42 @@ public int getTenantId() throws UserStoreException {
         return values.toArray(new String[values.size()]);
     }
 
+    protected String convertBytesToHexString(byte[] bytes) {
+        final StringBuilder builder = new StringBuilder();
+        for (byte b : bytes) {
+            builder.append("\\").append(String.format("%02x", b));
+        }
+        return builder.toString();
+    }
+
+    protected String transformUUIDToObjectGUID(String value) {
+        ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+        bb.putLong(java.util.UUID.fromString(value).getMostSignificantBits());
+        bb.putLong(java.util.UUID.fromString(value).getLeastSignificantBits());
+        final byte[] bytes = swapBytes(bb.array());
+        return convertBytesToHexString(bytes);
+    }
+
+    protected byte[] swapBytes(byte[] bytes) {
```

Comment:
This is a swapping operation specific to objectGUID right?
Shouldn't we name the method to reflect that?